### PR TITLE
MNIST and FashionMNIST now have their own 'raw' and 'processed' folders

### PR DESCRIPTION
**Pull request by dizcza**
Addresses https://github.com/pytorch/vision/issues/599.

### CODE REVIEW

File: torchvision/datasets/mnist.py [hunk 1]
```diff
@@ -95,21 +103,14 @@ def download(self):
...
-        # download files
-        try:
-            os.makedirs(os.path.join(self.root, self.raw_folder))
-            os.makedirs(os.path.join(self.root, self.processed_folder))
-        except OSError as e:
-            if e.errno == errno.EEXIST:
-                pass
-            else:
-                raise
+        os.makedirs(self.raw_folder, exist_ok=True)
```
Comment (fmassa) [inline torchvision/datasets/mnist.py]: `exist_ok` is Python 3 only. Can you make this Python 2 compatible?

File: torchvision/datasets/mnist.py [hunk 2]
```diff
@@ -220,30 +220,23 @@ def download(self):
         if self._check_exists():
             return
 
-        # download files
-        try:
```
Comment (fmassa) [inline torchvision/datasets/mnist.py]: I think it's better to keep this way of handling existing dirs, or maybe factor this out in a helper function.

File: torchvision/datasets/mnist.py (hunk 1 above)
Comment (dizcza) [reply torchvision/datasets/mnist.py]: How about `if not os.path.exist()`?

Comment (dizcza) [reply torchvision/datasets/mnist.py]: Ok, will do

**Comment by dizcza** [comment]
Ready.

File: torchvision/datasets/mnist.py [hunk 3]
```diff
@@ -3,11 +3,11 @@
 from PIL import Image
 import os
 import os.path
-import errno
+import gzip
```
[the remark this answers is not in the document]
Comment (dizcza) [reply torchvision/datasets/mnist.py]: Isn't gzip a built-in module?

[the remark this answers is not in the document]
Comment (fmassa) [reply torchvision/datasets/mnist.py]: Oh, looks like you are right. This is good then!

**Pull request merged**